### ISSUE TITLE
Streaming block production + Soft-commits

### ISSUE THREAD

**Issue by Voxelot** [FuelLabs/fuel-core#1206]
Rather than delaying transaction execution until a block interval is reached, we should consider moving towards a streaming architecture that allows transactions to be included into a block as soon as they reach the mempool. This would allow for soft-confirmations and improved latency. The block will be committed once the gas limit / byte size limit is reached, or after some kind of timeout to ensure liveness for syncing. Soft-confirmations will increase gossip overhead, but only by a factor of 2 typically. Once for the initial submission broadcast, and then another gossip broadcast once the tx is validated into a block (before the block is fully committed).

In a tendermint configuration, the producers may have to pause the mempool tx stream while the block gets ratified. However this streaming approach could still slightly improve latency for users who are lucky enough to land their transactions during the execution window.

cc: @SilentCicero 

- [ ] #1229

**Comment by MitchTurner** [comment]
Is this still something we want for Mainnet? Seems pretty low priority.

**Comment by MitchTurner** [comment]
@xgreenx I know you've been looking into pre-confirmations. Is that what this is?

**Comment by xgreenx** [comment]
Yes, I think we can close it now, since pre confirmations are implemented=)
